fog/view/server/src/sharding_strategy.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Enables a Fog View Store to know for which blocks to process TxOuts.
+//!
+//! By determining which TxOuts to process, we are able to "shard" the set of
+//! TxOuts across Fog View Store instances.
+
+use mc_blockchain_types::BlockIndex;
+use mc_fog_types::common::BlockRange;
+
+/// Tells a Fog View Store for which blocks it should process TxOuts.
+pub trait ShardingStrategy {
+    /// Returns true if the Fog View Store should process this block.
+    fn should_process_block(&self, block_index: BlockIndex) -> bool;
+}
+
+/// Determines whether or not to process a block's TxOuts based on the "epoch"
+/// sharding strategy, in which a block is processed IFF it falls within the
+/// contiguous range of blocks.
+///
+/// In practice, the set of Fog View Shards will contain overlapping
+/// [epoch_block_ranges] in order to obfuscate which shard processed the TxOuts.
+pub struct EpochShardingStrategy {
+    /// If a block falls within this range, then the Fog View Store should
+    /// process its TxOuts.
+    epoch_block_range: BlockRange,
+}
+
+impl ShardingStrategy for EpochShardingStrategy {
+    fn should_process_block(&self, block_index: BlockIndex) -> bool {
+        self.epoch_block_range.contains(block_index)
+    }
+}
+
+impl Default for EpochShardingStrategy {
+    fn default() -> Self {
+        Self {
+            epoch_block_range: BlockRange::new(0, u64::MAX),
+        }
+    }
+}
+
+impl EpochShardingStrategy {
+    #[allow(dead_code)]
+    pub fn new(epoch_block_range: BlockRange) -> Self {
+        Self { epoch_block_range }
+    }
+}
+
+#[cfg(test)]
+mod epoch_sharding_strategy_tests {
+    use super::*;
+
+    #[test]
+    fn should_process_block_block_index_is_before_epoch_start_returns_false() {
```

Comment:
Can these names be shorter, e.g. the name of this in the log is going to be `sharding_strategy::epoch_sharding_strategy_tests::should_process_block_block_index_is_before_epoch_start_returns_false`?

Like, `sharding::epoch::tests::no_block_index_before_epoch_start` is probably long enough, right?

Reply:
I follow the coneventions established [here](https://testing.googleblog.com/2007/02/tott-naming-unit-tests-responsibly.html) when naming unit tests. I always write the method under test, parameters / set up that makes the test unique, and then the expected result. While it may be shorter logs, it's less descriptive and the failure is harder to decipher. That said I don't feel particularly strongly about this, so would be happy to change it.

Reply:
I appreciate the intent of those conventions, but I think there's a couple problems with it.

On the one hand, it doesn't really help actual practice: when a test fails, I'm going to read the test anyways, so I can just read the test and parameters. It's just a missed opportunity for self-documentation, in the proper "comments should describe why the code is more than what it is" sense.

The second issue is that it's also just not very rustic, and the really long identifiers seem out of place with the rest of the ecosystem---e.g. it's `core::mem::ptr`, not `rust_core_library::memory_handling::pointer`.

Reply:
As discussed offline, we'll address this in a follow-up PR and take some time to discuss team conventions around naming unit tests.